modules/library/main/src/main/java/org/geotools/filter/function/StaticGeometry.java

```diff
@@ -477,8 +478,9 @@ public static String strStripAccents(String s) {
     }
 
     public static boolean strMatches(String s1, String s2) {
-        if (s1 == null || s2 == null) return false;
-        return s1.matches(s2);
+        if (s1 == null || s1.isEmpty() || s2 == null || s2.isEmpty()) return false;
```

Comment:
Should an empty string match an empty string?

Reply:
yes, probably.